subprojects/build-cache/src/main/java/org/gradle/caching/internal/controller/BuildCacheController.java

```diff
@@ -41,4 +41,6 @@ public interface BuildCacheController extends Closeable {
     Optional<BuildCacheLoadResult> load(BuildCacheKey cacheKey, CacheableEntity cacheableEntity);
 
     void store(BuildCacheKey cacheKey, CacheableEntity entity, Map<String, FileSystemSnapshot> snapshots, Duration executionTime);
+
+    Optional<String> getCacheSalt();
```

Comment:
:x: Would be great if we could avoid changing the existing API. But if we have to, let's make sure we add some documentation about this being a temporary measure.

Reply:
Another option would be to return a `Hasher` here. The default variant could just create one, and the NG variant could feed it some seed first.